### ISSUE TITLE
Expanded rows become hidden within the table

### ISSUE THREAD

**Issue by minuz** [TanStack/table#4970]
### Describe the bug

When expanding a row, the new added rows will hide some of the content, if not all. 
It seems that the `pageSize` has some impact on what's currently being rendered. 

### Your minimal, reproducible example

https://github.com/TanStack/table/tree/main/examples/react/expanding

### Steps to reproduce

1. Run the example `expanding`
2. Expand last row
3. Sub rows won't be visible 

### Expected behavior

As a user, I expect to be able to see the sub-rows, regardless the page size limit.

### How often does this bug happen?

Every time

### Platform

- macOS

### react-table version

8.9.3

### TypeScript version

5.0.4

### Additional context

Also using 
    - `@tanstack/react-query": "^4.29.13",`
    - `@tanstack/react-query-devtools": "^4.29.13"`
    - `@tanstack/react-table": "^8.9.3"`
    - `@tanstack/react-virtual": "^3.0.0-beta.54"`

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by DogAndHerDude** [comment]
Yeah, can second this. Similar setup but without `react-virtual`. Looked into hacking together a solution, it's very fiddly with the page size and expanded state, don't recommend managing that by counting the subrows and doing `pageSize + totalSubrowCount` it just very clunky.

However, I've noticed that in the demo for expanding rows it works fine. The setup is pretty similar to what I have, but for some reason it actually works.

**Comment by DogAndHerDude** [comment]
Turns out setting `paginateExpandedRows={false}` fixed the issue.

**Comment by Chaitanya-Upadhye** [comment]
> Turns out setting `paginateExpandedRows={false}` fixed the issue.

Yep, this worked for me
